cmd/booster-http/server.go

```diff
@@ -77,6 +84,55 @@ func (s *HttpServer) Stop() error {
 	return s.server.Close()
 }
 
+func (s *HttpServer) handleByPayloadCid(w http.ResponseWriter, r *http.Request) {
+	prefixLen := len(s.payloadBasePath())
+	if len(r.URL.Path) <= prefixLen {
+		msg := fmt.Sprintf("path '%s' is missing piece CID", r.URL.Path)
+		writeError(w, r, http.StatusBadRequest, msg)
+		return
+	}
+
+	payloadCidStr := r.URL.Path[prefixLen:]
+	payloadCid, err := cid.Parse(payloadCidStr)
+	if err != nil {
+		msg := fmt.Sprintf("parsing payload CID '%s': %s", payloadCidStr, err.Error())
+		writeError(w, r, http.StatusBadRequest, msg)
+		return
+	}
+
+	pieces, err := s.api.PiecesContainingMultihash(payloadCid.Hash())
+	if err != nil {
+		if isNotFoundError(err) {
+			msg := fmt.Sprintf("getting piece that contains payload CID '%s': %s", payloadCid, err.Error())
+			writeError(w, r, http.StatusNotFound, msg)
+			return
+		}
+		log.Errorf("getting piece that contains payload CID '%s': %s", payloadCid, err)
+		msg := fmt.Sprintf("server error getting piece that contains payload CID '%s'", payloadCidStr)
+		writeError(w, r, http.StatusInternalServerError, msg)
+		return
+	}
+
+	// Just get the content of the first piece returned (if the client wants a
+	// different piece they can just call the /piece endpoint)
+	pieceCid := pieces[0]
```

Comment:
Am I reading this code right in that this isn't actually restricted to the payload CID? So I could provide any indexed multihash and get back _some_ CAR that contains it?

Reply:
Yep that's right 👍